tests/build_tests/CMakeLists.txt

```diff
@@ -69,6 +69,12 @@ ENDIF()
 
 FOREACH(_step_full ${_steps})
   GET_FILENAME_COMPONENT(_step ${_step_full} NAME)
+  # Skip real-value only PETSc tests:
+  IF (DEAL_II_PETSC_WITH_COMPLEX)
+    IF ("${_step}" MATCHES "^(step-55|step-40|step-18|step-17)$")
+      CONTINUE ()
```

Comment:
judging from 
```
# Try to configure and build every example step in debug and release
# configuration. Error condition is that if a test is successfully
# configured it must compile successfully...
```
i would say these (`with_trilinos`, `with_mpi`) are not needed, as relevant tests would not configure. 
This implies that probably a better solution is to modify `CMakeLists.txt` of those tests which fail with complex petsc. This should give the same result, i will try...